scripts/build-tools.sh

```diff
@@ -5,27 +5,71 @@
 # fail immediately on any errors
 set -e
 
-cd tools
-
-rm -rf build_tools
-
-mkdir build_tools
-cd build_tools
-cmake ..
-make -j$(nproc)
-for args in $@
-do
-	#build test topologies
-	if [[ "$args" == "-t" ]]; then
-		make tests -j$(nproc)
-	fi
-	#build fuzzer
-	if [[ "$args" == "-f" ]]; then
-		rm -rf build_fuzzer
-		mkdir build_fuzzer
-		cd build_fuzzer
-		cmake ../../fuzzer
-		make -j$(nproc)
-		cd ../
-	fi
-done
+SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)
+SOF_REPO=$(dirname "$SCRIPT_DIR")
+NO_PROCESSORS=$(nproc)
+
+print_usage()
+{
+        echo "usage: $0 [-t|-f]"
+        echo "       [-t] Build test topologies"
+        echo "       [-f] Build fuzzer"
+}
+
+build_tools()
+{
+        cd "$SOF_REPO/tools"
+        rm -rf build_tools
+        mkdir build_tools
+        cd build_tools
+        cmake ..
+        make -j "$NO_PROCESSORS"
+}
+
+build_test()
+{
+        cd "$SOF_REPO/tools/build_tools"
+        make tests -j "$NO_PROCESSORS"
+}
+
+build_fuzzer()
+{
+        cd "$SOF_REPO/tools/build_tools"
+        rm -rf build_fuzzer
```

Comment:
This does nothing because `build_tools()` cleaned `build_tools/` a moment ago.  It's harmless except it gives the wrong impression `build_fuzzer()` can be invoked alone, but it cannot.